keanu-project/src/test/java/io/improbable/keanu/network/ModelCompositionTest.java

```diff
@@ -0,0 +1,126 @@
+package io.improbable.keanu.network;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import io.improbable.keanu.algorithms.variational.optimizer.gradient.GradientOptimizer;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.DoubleProxyVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.ParetoVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.UniformVertex;
+
+public class ModelCompositionTest {
+
+    private BayesianNetwork innerNet;
+    private UniformVertex trueLocation;
+    private BayesianNetwork outerNet;
+    private DoubleVertex location;
+    private DoubleVertex size;
+    private DoubleVertex gaussian;
+    private DoubleVertex pareto;
+    private DoubleVertex gaussOutputVertex;
+    private DoubleVertex paretoOutputVertex;
+
+    @Before
+    public void setup() {
```

Comment:
some comments here would be helpful, to describe the structure of the graph.

Reply:
I'll add a short comment.

Reply:
Looks great, thanks!